includes/discovery/functions.inc.php

```diff
@@ -812,6 +812,9 @@ function discover_entity_physical(&$valid, $device, $entPhysicalIndex, $entPhysi
                 'entPhysicalAlias'        => $entPhysicalAlias,
                 'entPhysicalAssetID'      => $entPhysicalAssetID,
             );
+            if (!empty($ifIndex)) {
+                $update_data['ifIndex'] = $ifIndex;
```

Comment:
You need to include ifIndex every time, otherwise it won't be unset if needed.

Reply:
Makes sense, of course.

Reply:
done